packages/cmd/src/CommandProcessor.ts

```diff
@@ -721,6 +777,115 @@ export class CommandProcessor {
 
     }
 
+    /**
+     * Print parameters in place for --dry-run in effect
+     * @private
+     * @param {IHandlerParameters} commandParameters
+     * @returns
+     * @memberof CommandProcessor
+     */
+    private showInputsOnly(response: IHandlerResponseApi, commandParameters: IHandlerParameters) {
+
+        /**
+         * Determine if we should display secure values.  If the ENV variable is set to true,
+         * then we will display the secure values.  If the ENV variable is set to false,
+         * then we will display the non-secure values.
+         *                                                             - GitHub Copilot AI
+         */
+        let showSecure = false;
+
+        // ZOWE_SHOW_SECURE_ARGS
+        const SECRETS_ENV = `${ImperativeConfig.instance.envVariablePrefix}${CommandProcessor.ENV_SHOW_SECURE_SUFFIX}`;
+        const env = (process.env[SECRETS_ENV] || "false").toUpperCase();
+
+        if (env === "TRUE" || env === "1") {
+            showSecure = true;
+        }
+
+        // if config exists and a layer exists
+        let useConfig = false;
+        this.mConfig?.layers.forEach((layer) => {
+            if (layer.exists) {
+                useConfig = true;
+            }
+        });
+
+        /**
+         * Determine if Zowe V2 Config is in effect.  If it is, then we will construct
+         * a Set of secure fields from its API.  If it is not, then we will construct
+         * a Set of secure fields from the `ConnectionPropsForSessCfg` defaults.
+         */
+        const secureInputs: Set<string> =
+            useConfig ?
+                new Set(this.mConfig.api.secure.secureFields().map(v => v.split('.').slice(-1)[0])) :
```

Comment:
I think `this.mConfig.api.secure.securePropsForProfile` should be used here instead. Getting the list of all secure fields in the file may have undesired behavior. For example with the following config:
```
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "lpar1": {
            "type": "zosmf",
            "properties": {
                "host": "lpar1.broadcom.net",
                "port": 443
            },
            "secure": []
        },
        "lpar2": {
            "type": "zosmf",
            "properties": {
                "port": 1443
            },
            "secure": [
                "host"
            ]
        },
        "base": {
            "type": "base",
            "properties": {
                "rejectUnauthorized": true
            },
            "secure": [
                "user",
                "password"
            ]
        }
    },
    "defaults": {
        "zosmf": "lpar1",
        "base": "base"
    },
    "autoStore": true
}
```

Here the host property is only secure in the lpar2 profile, so running `zowe jobs ls jobs --show-inputs-only` should show the host property in plain text that is present in the default lpar1 profile. Currently I see `host: (secure value)` instead.

Reply:
your point makes sense - i suppose the same will happen if a base profile marks `host` secure while zosmf does not; i think `host` will be in the secure list.  the undesired behavior is that we may mistakenly show something as `(secure value)` when it is not.

one struggle i have - i'm not clear how to determine the name of the profile used from `CommandProcessor`.  i think we had this problem before in command handlers: we have access to the profile contents / values and type but not the profile name to use `securePropsForProfile`.  do you see some way to determine this?

Reply:
Sorry I missed this comment until now, I believe `ConfigUtils.getActiveProfileName` might help here: https://github.com/zowe/imperative/blob/next/packages/config/src/ConfigUtils.ts#L49

Reply:
side note, i noticed, `ConfigUtils.getActiveProfileName` returns `host` as a secure prop for `zosmf2` even though it is not:
```
        "lpar1": {
            "properties": {
                "host": "usilca11.lvn.broadcom.net"
            },
            "profiles": {
                "zosmf": {
                    "type": "zosmf",
                    "properties": {
                        "port": 1443
                    },
                    "secure": [
                        "host"
                    ]
                },
                "zosmf2": {
                    "type": "zosmf",
                    "properties": {
                        "port": 1443
                    },
                    "secure": []
                },
```

in any case, this is only a display command option to help users.  the only impact in worst case scenario is that we show a field as `(secure value)` when it is not.  it also seems unlikely in both of our examples that a user would want a field like `host` secure for one `zosmf` profile and not another.  

no action needed - i'm just recording considerations for this discussion

Reply:
I thought that a PR was recently merged which resolved an issue about returning host as secure when it is only secure for some profiles. Perhaps after you merge next into this branch, the problem will go away?

Reply:
I believe the latest changes from next have been merged in, if this behavior persists I think we should file it as an issue. 🙂

Reply:
correction, it's `this.mConfig.api.secure.securePropsForProfile` instead of `ConfigUtils.getActiveProfileName` that causes an issue.  

it still happens to me after syncing with `next` this morning.  i dont get an issue using @t1m0thyj's sample profile with `lpar1.zosmf` and `lpar2.zosmf`, but i do get an issue with `lpar1.zosmf` and `lpar1.zosmf2` as in my example.  

two instances of zosmf on the sample lpar2 don't make sense, but i the `lpar` higher level is an arbitrary construction.  

Here is the code with trace messages added:
```typescript
    public securePropsForProfile(profileName: string) {
        console.log(`>>>> entry <<<<`)
        const profilePath = this.mConfig.api.profiles.expandPath(profileName);
        console.log(`profilePath: ${profilePath}`)
        const secureProps = [];
        for (const propPath of this.secureFields()) {
            console.log(`propPath: ${propPath}`)
            const pathSegments = propPath.split(".");  // profiles.XXX.properties.YYY
            // eslint-disable-next-line @typescript-eslint/no-magic-numbers
            if (profilePath.startsWith(pathSegments.slice(0, -2).join("."))) {
                secureProps.push(pathSegments.pop());
            }
        }
        console.log(`secureProps ${secureProps}`)
        console.log(`>>>> exit <<<<`)
        return secureProps;
    }
```

here's the command i run and output I see:
```
C:\dev\systems\ca11>zowe jobs list jobs --show-inputs-only --zosmf-p lpar1.zosmf2
>>>> entry <<<<
profilePath: profiles.lpar1.profiles.zosmf2
propPath: profiles.lpar1.profiles.zosmf.properties.host
propPath: profiles.my_base.properties.user
propPath: profiles.my_base.properties.password
secureProps host
>>>> exit <<<<
>>>> entry <<<<
profilePath: profiles.my_base
propPath: profiles.lpar1.profiles.zosmf.properties.host
propPath: profiles.my_base.properties.user
propPath: profiles.my_base.properties.password
secureProps user,password
>>>> exit <<<<
```

it looks like `zosmf2` becomes `zosmf` in `ConfigProfiles.expandPath`